packages/zoe/src/zoeService/zoe.js

```diff
@@ -62,7 +62,8 @@ const makeZoeKit = (
     shutdownZoeVat,
   );
 
-  const getBundleCapFromID = bundleID => E(vatAdminSvc).getBundleCap(bundleID);
+  const getBundleCapFromID = bundleID =>
+    E(vatAdminSvc).waitForBundleCap(bundleID);
```

Comment:
@warner This was the most surgical change I could make to ensure that `E(zoe).install(hashBundle)` would wait for the chain to receive the bundle separately. This might be giving us a hint that `getBundleCap` should behave like `waitForBundleCap`, rather than having separate methods, or it might be a hint that `getBundleCapFromID` defined here should be renamed `waitForBundleCapWithID` or some such.

Reply:
I think this is fine. I vaguely remember thinking that Zoe's `E(zoe).install()` should wait for the bundle to be installed out-of-band, so that you can only get an installation handle for installations that are actually functional, and aren't waiting for a bundle install that might never happen. And since `E(zoe).install()` is interacting with a wrapper function (which sometimes comes from the fake vatadmin service), that wrapper function should do the waiting, which means that when it calls the real vatadmin service, it should be using `waitForBundleCap` instead of `getBundleCap`.

I've assumed that other (non-zoe, non-agoric) uses/users of swingset and `vatAdminService` may have legitimate reasons to query for a bundleID and want a negative answer (instead of an unbounded pause), which means keeping `getBundleCap` as the form that doesn't wait, and Zoe can use `waitForBundleCap` (which does).

And yeah, at some point renaming this zoe thing `waitForBundleCapFromID` might make sense.

Incidentally, while investigating this function, I found it is named `getBundleCapFromID` here and in `installationStorage.js`, but `getBundleCapForID` in `zoe.js`, which is a bit confusing for code spelunkers.